aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -2011,29 +2012,27 @@ static void check_str_ord_valid(const std::string& str_ord, optional<IntArrayRef
 
 // Performs vector norm for ord = +/-infinity, and the second dimension reduction
 // for matrix norms.
-static Tensor _norm_min_max(Tensor& self, double ord, int64_t dim, bool keepdim) {
-  Tensor result;
-  if (self.numel() == 0 && self.sizes()[dim] > 0) {
+static Tensor _norm_min_max(Tensor& result, double ord, int64_t dim, bool keepdim) {
+  if (result.numel() == 0 && result.sizes()[dim] > 0) {
     // This special case is needed in matrix norm for tensors with 3 or more dims,
     // or in vector norm for order inf and -inf for tesnsors with 2 or more dims.
     // When the sizes of the dims to be reduced are greater than 0 but another dim
     // in the tensor is size 0 (thus numel == 0), we must either flatten or resize
     // the second reduction dim to 1, to avoid calling min/max, which would throw
     // an error.
-    if (self.sizes()[dim] != 1) {
-      auto new_sizes = self.sizes().vec();
+    if (result.sizes()[dim] != 1) {
```

Comment:
@kurtamohler would you like to sanity check these norm changes?

Reply:
These changes renamed `self` as `result` to be consistent with the rest of the codebase. I also did some early returns, to remove the `Tensor result;` variable to avoid having to rename it.
As a side note, `result.sizes()[dim]` can be written more sucintly as `result.size(dim)`, but I preferred not to go down the rabbit hole...

Reply:
Nevermind, I just reverted these changes to reduce the surface of the PR as suggested by @mruberry